File: app/src/main/java/dev/ragnarok/fenrir/api/interfaces/IDocsApi.java
package dev.ragnarok.fenrir.api.interfaces;

import androidx.annotation.CheckResult;

import java.util.Collection;
import java.util.List;

import dev.ragnarok.fenrir.api.model.AccessIdPair;
import dev.ragnarok.fenrir.api.model.Items;
import dev.ragnarok.fenrir.api.model.VkApiDoc;
import dev.ragnarok.fenrir.api.model.server.VkApiDocsUploadServer;
import dev.ragnarok.fenrir.api.model.server.VkApiVideosUploadServer;
import io.reactivex.rxjava3.core.Single;


public interface IDocsApi {

    @CheckResult
    Single<Boolean> delete(Integer ownerId, int docId);

    @CheckResult
    Single<Integer> add(int ownerId, int docId, String accessKey);

    @CheckResult
    Single<List<VkApiDoc>> getById(Collection<AccessIdPair> pairs);

    @CheckResult
    Single<Items<VkApiDoc>> search(String query, Integer count, Integer offset);

    @CheckResult
    Single<VkApiDoc.Entry> save(String file, String title, String tags);

    @CheckResult
    Single<VkApiDocsUploadServer> getUploadServer(Integer groupId);

    @CheckResult
    Single<VkApiDocsUploadServer> getMessagesUploadServer(Integer peerId, String type);

    @CheckResult
    Single<VkApiVideosUploadServer> getVideoServer(Integer isPrivate, Integer group_id, String name);

    @CheckResult
    Single<Items<VkApiDoc>> get(Integer ownerId, Integer count, Integer offset, Integer type);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/interfaces/IDocsApi.kt
package dev.ragnarok.fenrir.api.interfaces

import androidx.annotation.CheckResult
import dev.ragnarok.fenrir.api.model.AccessIdPair
import dev.ragnarok.fenrir.api.model.Items
import dev.ragnarok.fenrir.api.model.VKApiDoc
import dev.ragnarok.fenrir.api.model.server.VKApiDocsUploadServer
import dev.ragnarok.fenrir.api.model.server.VKApiVideosUploadServer
import io.reactivex.rxjava3.core.Single

interface IDocsApi {
    @CheckResult
    fun delete(ownerId: Int?, docId: Int): Single<Boolean>

    @CheckResult
    fun add(ownerId: Int, docId: Int, accessKey: String?): Single<Int>

    @CheckResult
    fun getById(pairs: Collection<AccessIdPair>): Single<List<VKApiDoc>>

    @CheckResult
    fun search(query: String?, count: Int?, offset: Int?): Single<Items<VKApiDoc>>

    @CheckResult
    fun save(file: String?, title: String?, tags: String?): Single<VKApiDoc.Entry>

    @CheckResult
    fun getUploadServer(groupId: Int?): Single<VKApiDocsUploadServer>

    @CheckResult
    fun getMessagesUploadServer(peerId: Int?, type: String?): Single<VKApiDocsUploadServer>

    @CheckResult
    fun getVideoServer(
        isPrivate: Int?,
        group_id: Int?,
        name: String?
    ): Single<VKApiVideosUploadServer>

    @CheckResult
    operator fun get(
        ownerId: Int?,
        count: Int?,
        offset: Int?,
        type: Int?
    ): Single<Items<VKApiDoc>>
}